File: sdk/cpp/examples/02_cpp_main_api/students_test.cpp
// Copyright 2021 Touca, Inc. Subject to Apache-2.0 License.

#include "students.hpp"

#include "students_types.hpp"
#include "touca/touca.hpp"

int main(int argc, char* argv[]) {
  touca::workflow("find_student", [](const std::string& username) {
    const auto& student = find_student(username);
    touca::assume("username", student.username);
    touca::check("fullname", student.fullname);
    touca::check("birth_date", student.dob);
    touca::check("gpa", student.gpa);
    touca::add_metric("external_source", 1500);
  });
  touca::run(argc, argv);
}

File: examples/js/02_node_main_api/students_test.ts
// Copyright 2022 Touca, Inc. Subject to Apache-2.0 License.

import touca from '@touca/node';
import { find_student } from './students.js';

touca.add_serializer('Date', (x) => {
  const d = x as Date;
  return {
    day: d.getUTCDate(),
    month: d.getUTCMonth(),
    year: d.getUTCFullYear()
  };
});

touca.workflow('students', async (username: string) => {
  touca.start_timer('find_student');
  const student = await find_student(username);
  touca.stop_timer('find_student');
  touca.assume('username', student.username);
  touca.check('fullname', student.fullname);
  touca.check('birth_date', student.dob);
  touca.check('gpa', student.gpa);
  touca.check('pass', student.gpa < 3.9);
  touca.add_metric('external_source', 1500);
});

touca.run();
